Backpack-SwiftUI/TextArea/Classes/BPKTextArea.swift

```diff
@@ -0,0 +1,136 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright © 2023 Skyscanner Ltd. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+import UIKit
+
+public struct BPKTextArea: View {
+    
+    private enum TextEditorConstants {
+        static let verticalPadding: BPKSpacing = .md
+        static let horizontalPadding: CGFloat = 12
+    }
+    
+    private enum PlaceholderConstants {
+        static let verticalPadding: BPKSpacing = .base
+        static let horizontalPadding: BPKSpacing = .base
+    }
+    
+    private enum BorderConstants {
+        static let cornerRadius: BPKCornerRadius = .xs
+        static let lineWidth: CGFloat = 1
+    }
+    
+    @Environment(\.colorScheme) var colorScheme
+    @Binding private var value: String
+    private let charLimit: Int?
+    private let placeholder: String
+    private var onValueChange: (String) -> Void = { _ in }
+    
+    /// Creates a `BPKTextArea`.
+    ///
+    /// - Parameters:
+    ///   - text: The text to display in the text field.
+    ///   - placeholder: The placeholder text to display when the text field is empty.
+    ///   - charLimit: Limit of character allowed to enter
+    ///   - onValueChange: Callback
+    public init(
+        _ value: Binding<String>,
+        placeholder: String = "",
```

Comment:
I think the placeholder should either be nullable or not have a default empty value, it should be clear to the client of this api that the component requires a placeholder (or not).

Reply:
Thanks for spotting. Placeholder is made optional.